src/backend/distributed/utils/enable_ssl.c

```diff
@@ -73,36 +72,39 @@ citus_setup_ssl(PG_FUNCTION_ARGS)
 	ereport(WARNING, (errmsg("can not setup ssl on postgres that is not compiled with "
 							 "ssl support")));
 #else /* USE_SSL */
-	if (!EnableSSL && ShouldUseAutoSSL())
+	if (EnableSSL)
 	{
-		ereport(LOG, (errmsg("citus extension created on postgres without ssl enabled, "
-							 "turning it on during creation of the extension")));
-
-		/* execute the alter system statement to enable ssl on within postgres */
-		Node *enableSSLParseTree = ParseTreeNode(ENABLE_SSL_QUERY);
-		AlterSystemSetConfigFile((AlterSystemStmt *) enableSSLParseTree);
-
-		if (strcmp(SSLCipherSuites, POSTGRES_DEFAULT_SSL_CIPHERS) == 0)
-		{
-			/*
-			 * postgres default cipher suite is configured, these allow TSL 1 and TLS 1.1,
-			 * citus will upgrade to TLS1.2+HIGH and above.
-			 */
-			Node *citusSSLCiphersParseTree = ParseTreeNode(SET_CITUS_SSL_CIPHERS_QUERY);
-			AlterSystemSetConfigFile((AlterSystemStmt *) citusSSLCiphersParseTree);
-		}
+		/* It's already on */
+		PG_RETURN_NULL();
+	}
 
+	ereport(LOG, (errmsg("citus extension created on postgres without ssl enabled, "
+						 "turning it on during creation of the extension")));
+
+	/* execute the alter system statement to enable ssl on within postgres */
+	Node *enableSSLParseTree = ParseTreeNode(ENABLE_SSL_QUERY);
+	AlterSystemSetConfigFile((AlterSystemStmt *) enableSSLParseTree);
+
+	if (strcmp(SSLCipherSuites, POSTGRES_DEFAULT_SSL_CIPHERS) == 0)
+	{
 		/*
-		 * ssl=on requires that a key and certificate are present, since we have
-		 * enabled ssl mode here chances are the user didn't install credentials already.
-		 *
-		 * This function will check if they are available and if not it will generate a
-		 * self singed certificate.
+		 * postgres default cipher suite is configured, these allow TSL 1 and TLS 1.1,
+		 * citus will upgrade to TLS1.2+HIGH and above.
 		 */
-		CreateCertificatesWhenNeeded();
-
-		GloballyReloadConfig();
+		Node *citusSSLCiphersParseTree = ParseTreeNode(SET_CITUS_SSL_CIPHERS_QUERY);
+		AlterSystemSetConfigFile((AlterSystemStmt *) citusSSLCiphersParseTree);
 	}
+
+	/*
+	 * ssl=on requires that a key and certificate are present, since we have
```

Comment:
I guess this is private key, so maybe `.. that a server private key and server certificate are present ...`